test/lib.bash

```diff
@@ -202,14 +202,10 @@ function run_knative_serving_rolling_upgrade_tests {
       # Check we got RequirementsNotMet error
       [[ $(oc get ClusterServiceVersion $upgrade_to -n $OPERATORS_NAMESPACE -o=jsonpath="{.status.requirementStatus[?(@.name==\"$upgrade_to\")].message}") =~ "requirement not met: minKubeVersion" ]] || return 1
       # Check KnativeServing still has the old version
-      [[ $(oc get knativeserving.serving.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") == "$serving_version" ]] || return 1
+      [[ $(oc get knativeserving knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") == "$serving_version" ]] || return 1
     else
       approve_csv "$upgrade_to" || return 1
-      # If we used old CR for installing Serverless both knativeserving CRs should be updated now
-      if oc get knativeserving.serving.knative.dev knative-serving -n "$SERVING_NAMESPACE" >/dev/null 2>&1; then
-        timeout 900 '[[ ! ( $(oc get knativeserving.serving.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") != $serving_version && $(oc get knativeserving.serving.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
-      fi
-      timeout 900 '[[ ! ( $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") != $serving_version && $(oc get knativeserving.operator.knative.dev knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
+      timeout 900 '[[ ! ( $(oc get knativeserving knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") != $serving_version && $(oc get knativeserving knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1
 
       # Assert that the old image references eventually fade away
       timeout 900 "oc get pod -n $SERVING_NAMESPACE -o yaml | grep image: | uniq | grep $serving_version" || return 1
```

Comment:
```suggestion
      # TODO: reenable once we do a Knative version upgrade
      # timeout 900 '[[ ! ( $(oc get knativeserving knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.version}") != $serving_version && $(oc get knativeserving knative-serving -n $SERVING_NAMESPACE -o=jsonpath="{.status.conditions[?(@.type==\"Ready\")].status}") == True ) ]]' || return 1

      # Assert that the old image references eventually fade away
      # timeout 900 "oc get pod -n $SERVING_NAMESPACE -o yaml | grep image: | uniq | grep $serving_version" || return 1
```

I think we can comment these checks out until we actually do a Knative upgrade.

Reply:
Is it useful? You'll need to remember to re-enable this. We could keep it as it is and merge pull requests manually until the upgrade properly works (hopefully very soon). Ideally, we'd really define the new CSV together with the new images.

Reply:
We can track reenablement via a JIRA. We actuially need smarter code here that detects if we're even installing a new version after all. A release might not entail a new Knative version (see patch releases for example).

Reply:
What about introducing an annotation on the CSV, something like `knative-update: "true"` , the test could check if knative-update is expected and change its behaviour?
I guess patch releases might and might not update Knative (both options are possible) and for changes like those in this PR we might as well have both options. So I think we need to set explicitly whether knative is expected to be updated. We can't detect that from e.g. the micro version in `version: 1.4.1`.

Anyway, I'm not sure disabling this code (and having JIRA to re-enable it) makes our life much easier.

Reply:
Ooookay, let's just merge this one then to unblock CSV dependent work?